# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.cpp
# Commit: 2a39a8e425b0
# Full Hash: 2a39a8e425b0824c670ba03452c437c3748c990b
# Author: Greg V <greg@unrelenting.technology>
# Date: 2020-01-06 21:54:03
# Regressor Bug: 1587360
# File Overlap Count: 1
# Description:
#   Bug 1587360 - Add support for idle-inhibit Wayland protocol r=stransky
#   
#   The idle-inhibit protocol allows idle wake locks on Wayland
#   without any of the supported D-Bus interfaces running.
#   
# ==============================================================================

diff -r 47592189b305 -r 2a39a8e425b0 widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp	Mon Jan 06 17:02:14 2020 +0000
+++ b/widget/gtk/nsWaylandDisplay.cpp	Mon Jan 06 16:11:19 2020 +0000
@@ -147,6 +147,11 @@
   mPrimarySelectionDeviceManager = aPrimarySelectionDeviceManager;
 }
 
+void nsWaylandDisplay::SetIdleInhibitManager(
+    zwp_idle_inhibit_manager_v1* aIdleInhibitManager) {
+  mIdleInhibitManager = aIdleInhibitManager;
+}
+
 void nsWaylandDisplay::SetDmabuf(zwp_linux_dmabuf_v1* aDmabuf) {
   mDmabuf = aDmabuf;
 }
@@ -238,6 +243,13 @@
     wl_proxy_set_queue((struct wl_proxy*)primary_selection_device_manager,
                        display->GetEventQueue());
     display->SetPrimarySelectionDeviceManager(primary_selection_device_manager);
+  } else if (strcmp(interface, "zwp_idle_inhibit_manager_v1") == 0) {
+    auto idle_inhibit_manager =
+        static_cast<zwp_idle_inhibit_manager_v1*>(wl_registry_bind(
+            registry, id, &zwp_idle_inhibit_manager_v1_interface, 1));
+    wl_proxy_set_queue((struct wl_proxy*)idle_inhibit_manager,
+                       display->GetEventQueue());
+    display->SetIdleInhibitManager(idle_inhibit_manager);
   } else if (strcmp(interface, "wl_compositor") == 0) {
     // Requested wl_compositor version 4 as we need wl_surface_damage_buffer().
     auto compositor = static_cast<wl_compositor*>(
@@ -388,6 +400,7 @@
       mShm(nullptr),
       mSyncCallback(nullptr),
       mPrimarySelectionDeviceManager(nullptr),
+      mIdleInhibitManager(nullptr),
       mRegistry(nullptr),
       mDmabuf(nullptr),
       mGbmDevice(nullptr),